phylokit/distance.py

```diff
@@ -0,0 +1,145 @@
+# Tree distance metrics.
+import numba
+import numpy as np
+
+
+def check_node_bounds(tree, *args):
+    """
+    Checks that the specified node is within the tree.
+
+    :param tskit.Tree tree: The tree to check.
+    :param int `*args`: The node IDs to check.
+    :raises ValueError: If any of the nodes are outside the tree.
+    """
+    num_nodes = tree.parent_array.shape[0] - 1
+    for u in args:
+        if u < 0 or u > num_nodes:
+            raise ValueError(f"Node {u} is not in the tree")
+
+
+@numba.njit(cache=True)
+def _branch_length(parent, time, u):
+    ret = 0
+    p = parent[u]
+    if p != -1:
+        ret = time[p] - time[u]
+    return ret
+
+
+def branch_length(tree, u):
```

Comment:
I don't think we want to export this as a function, as it's really an optional variable that we might store in the dataset (see [here](https://github.com/jeromekelleher/phylokit/blob/3f1c2d2fe6bb47035a6c9cee1a8a14565a9f8e97/phylokit/convert.py#L121))

I guess the simplest thing to do is leave things as they are for now and to change the code as part of #14 but to remove the branch_length function from the public API.

Reply:
I see, I'll do that in the next PR

Reply:
OK, merging.